format/spec.md

```diff
@@ -732,12 +738,10 @@ Valid snapshots are stored as a list in table metadata. For serialization, see A
 
 #### Snapshot Row IDs
 
-A snapshot's `first-row-id` is assigned to the table's current `next-row-id` on each commit attempt. If a commit is retried, the `first-row-id` must be reassigned. If a commit contains no new rows, `first-row-id` should be omitted.
+A snapshot's `first-row-id` is assigned to the table's current `next-row-id` on each commit attempt. If a commit is retried, the `first-row-id` must be reassigned based on the table's current `next-row-id`. The `first-row-id` field is required even if a commit does not assign any ID space.
```

Comment:
Always writing `first-row-id` distinguishes snapshots that added 0 rows from pre-v3 snapshots.